File: VB/GridDemo.Wpf/Controls/CustomClasses.vb
Imports DevExpress.Xpf.Grid
Imports DevExpress.Xpf.Utils
Imports System.Collections.Generic
Imports System.Windows
Imports System.Windows.Controls

Namespace GridDemo
    Public Class GridSummaryList
        Inherits List(Of GridSummaryItem)

    End Class
    Public Class NameTextControl
        Inherits Control

        Public Shared ReadOnly NameValueProperty As DependencyProperty = DependencyProperty.Register("NameValue", GetType(String), GetType(NameTextControl), New PropertyMetadata(Nothing))
        Public Shared ReadOnly TextValueProperty As DependencyProperty = DependencyProperty.Register("TextValue", GetType(String), GetType(NameTextControl), New PropertyMetadata(Nothing))
        Public Property NameValue() As String
            Get
                Return DirectCast(GetValue(NameValueProperty), String)
            End Get
            Set(ByVal value As String)
                SetValue(NameValueProperty, value)
            End Set
        End Property
        Public Property TextValue() As String
            Get
                Return DirectCast(GetValue(TextValueProperty), String)
            End Get
            Set(ByVal value As String)
                SetValue(TextValueProperty, value)
            End Set
        End Property
        Public Sub New()
            Me.SetDefaultStyleKey(GetType(NameTextControl))
        End Sub
    End Class
    Public Class HintControl
        Inherits ContentControl

        Public Sub New()
            Me.SetDefaultStyleKey(GetType(HintControl))
        End Sub
    End Class
End Namespace

File: CS/GridDemo.Wpf/Controls/CustomClasses.cs
using DevExpress.Xpf.Grid;
using DevExpress.Xpf.Utils;
using System.Collections.Generic;
using System.Windows;
using System.Windows.Controls;

namespace GridDemo {
    public class GridSummaryList : List<GridSummaryItem> { }
    public class NameTextControl : Control {
        public static readonly DependencyProperty NameValueProperty =
            DependencyProperty.Register("NameValue", typeof(string), typeof(NameTextControl), new PropertyMetadata(null));
        public static readonly DependencyProperty TextValueProperty =
            DependencyProperty.Register("TextValue", typeof(string), typeof(NameTextControl), new PropertyMetadata(null));
        public string NameValue {
            get { return (string)GetValue(NameValueProperty); }
            set { SetValue(NameValueProperty, value); }
        }
        public string TextValue {
            get { return (string)GetValue(TextValueProperty); }
            set { SetValue(TextValueProperty, value); }
        }
        public NameTextControl() {
            this.SetDefaultStyleKey(typeof(NameTextControl));
        }
    }
    public class HintControl : ContentControl {
        public HintControl() {
            this.SetDefaultStyleKey(typeof(HintControl));
        }
    }
}
